hal/inc/i2c_hal.h

```diff
@@ -52,6 +53,7 @@ typedef struct hal_i2c_config_t {
     uint32_t rx_buffer_size;
     uint8_t* tx_buffer;
     uint32_t tx_buffer_size;
+    uint8_t  freeable;
```

Comment:
Perhaps consider using `flags` instead of a single member variable for future extensibility.